fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/services/CredentialsService.java

```diff
@@ -25,11 +24,11 @@
 public interface CredentialsService {
 
     /**
-     * Get the credentials for the given servlet request
+     * Get the credentials for the given request
      *
-     * @param request the servlet request
+     * @param request the request
      * @return the JCR credentials for the given request
      */
-    Credentials getCredentials(final HttpServletRequest request);
+    Credentials getCredentials(final Object request);
```

Comment:
Basically, yes.

This `getCredentials` service was added in order to remove one more Modeshape dependency from the HTTP layer: https://github.com/fcrepo4/fcrepo4/commit/2d3506051df643e18da53a684fd1b81e77d2a7d8#diff-be95e9939194d194be91ae919de03353L148

As you know, I am keenly interested in removing JCR dependencies from the kernel-api and not adding to them, but a related step in that same direction involves removing MODE dependencies from the HTTP layers. In fact, that seems to be a necessary precondition (or at least a logical precondition).

I suspect that when we eventually replace the use of `jcr.Session` with a Fedora-specific one, this `jcr.Credentials` service will go away, as it should be pretty easy at that point to factor this method into a generic `FedoraSession` interface.